src/nebari_plugin_self_registration/terraform/modules/self-registration/main.tf

```diff
@@ -12,6 +12,11 @@ locals {
     enabled  = false
     selector = null
   }
+
+  affinity_selector_key = {
```

Comment:
@kenafoster I created a new local variable here instead of using `affinity` as I don't think this should be exposed to the user but let me know if you have a different opinion.

Reply:
I believe it makes sense to not expose the affinity key to the end user since that's error prone and cloud specific, but to expose the value - i.e. allow the user to specify a node pool other than 'general' for both the app and the job.

However, I also noticed that there's a typo in the previous implementation.  Class names need updating - I'll add comments - could you fix in your PR?

Here's an example of it implemented correctly: https://github.com/MetroStar/nebari-label-studio/blob/main/src/nebari_plugin_label_studio_chart/plugin.py#L13